compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1933,28 +1933,35 @@ TR::Register *OMR::X86::TreeEvaluator::integerDualMulEvaluator(TR::Node *node, T
 
 TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   TR::Register *targetRegister = 0;
-   TR::Node     *firstChild     = node->getFirstChild();
-   TR::Node     *secondChild    = node->getSecondChild();
-   bool          nodeIs64Bit    = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
-   intptrj_t    value;
-
+   TR::Register *targetRegister  = 0;
+   TR::Node     *firstChild      = node->getFirstChild();
+   TR::Node     *secondChild     = node->getSecondChild();
+   bool          nodeIs64Bit     = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
+   bool     oursIsTheOnlyMemRef  = true;
+   bool     countsAreDecremented = false;
+   bool firstChildRefCountIsZero = false;
+   TR::MemoryReference *tempMR   = NULL;
+   bool isMemRef                 = firstChild->getOpCode().isMemoryReference();
+   TR::DataType dataType         = secondChild->getDataType();
+   int64_t       value;
+   
    if (node->isDualCyclic())
       {
       return TR::TreeEvaluator::integerDualMulEvaluator(node, cg);
       }
-
-   if (secondChild->getOpCode().isLoadConst() &&
-      (value = TR::TreeEvaluator::integerConstNodeValue(secondChild, cg)))
+   
+   if(secondChild->getOpCode().isLoadConst())
       {
-      intptrj_t absValue = (value < 0) ? -value : value;
-      bool firstChildRefCountIsZero = false;
+      value = (dataType == TR::Int8) ? secondChild->getByte() :
+              (dataType == TR::Int16) ? secondChild->getShortInt() : TR::TreeEvaluator::integerConstNodeValue(secondChild, cg);
```

Comment:
On a second thought, we can use `integerConstNodeValue ` for all data types here. integerConstNodeValue will just cast the data to `intptrj_t`.